pkg/controller.v1/pytorch/hpa.go

```diff
@@ -45,21 +47,20 @@ func (r *PyTorchJobReconciler) ReconcileHPA(pytorchJob *kubeflowv1.PyTorchJob) e
 		return err
 	}
 
-	if err := r.Get(context.TODO(), types.NamespacedName{
-		Name:      pytorchJob.Name,
-		Namespace: pytorchJob.Namespace,
-	}, current); err != nil {
-		if !errors.IsNotFound(err) {
-			return err
-		}
-
-		// Create the new HPA.
-		logger.V(1).Info("Creating HPA", "namespace", expected.Namespace, "name", expected.Name)
-		err = r.Create(context.TODO(), expected)
-		if err != nil {
+	err = r.Get(context.TODO(), client.ObjectKeyFromObject(expected), current)
+	if err != nil {
+		if errors.IsNotFound(err) && !trainutil.IsJobSuspended(&pytorchJob.Spec.RunPolicy) {
```

Comment:
What if err == errors.IsNotFound and job is suspended ? What should it return? Nil or IsNotFound ?

Reply:
It's a good point.
I think we should return nil. Because if the job is suspended, it's correct that HPA doesn't exist.

Reply:
Done.